java/vector/src/main/java/org/apache/arrow/vector/util/DecimalUtility.java

```diff
@@ -89,13 +89,29 @@ public static byte[] getByteArrayFromArrowBuf(ArrowBuf bytebuf, int index, int b
    * thrown, otherwise returns true.
    */
   public static boolean checkPrecisionAndScale(BigDecimal value, int vectorPrecision, int vectorScale) {
+    return checkPrecisionAndScale(value, vectorPrecision, vectorScale, true);
+  }
+
+  /**
+   * Check that the BigDecimal scale equals the vectorScale and that the BigDecimal precision is
+   * less than or equal to the vectorPrecision. If not, then either ann UnsupportedOperationException is
+   * thrown or false is returned if depending on the shouldThrow parameter, otherwise returns true.
+   */
+  public static boolean checkPrecisionAndScale(BigDecimal value, int vectorPrecision, int vectorScale,
+                                               boolean shouldThrow) {
```

Comment:
`shouldThrow` is a bit of an awkward API; it seems like there's not that much logic here, and we could just duplicate it?

Reply:
Added checkPrecisionAndScaleNoThrow() to replace this overload.